### DISCUSSION TITLE
json alters dict keys type

### DISCUSSION THREAD

**Discussion by kjm1102** [micropython/micropython#12300] [Using MicroPython]
Why does json change my dictionary keys from integers to strings?
```
import json; dic={}
for i in range(2): dic[i]=[i, str(i)]
print(dic)
with open('dic', 'w') as f: json.dump(dic,f)
with open('dic', 'r') as f: print(json.load(f))

{0: [0, '0'], 1: [1, '1']}
{'1': [1, '1'], '0': [0, '0']}

>>> import sys; sys.implementation
[object_link_for_thonny=1061265408](name='micropython', version=(1, 20, 0), _machine='LILYGO TTGO LoRa32 with ESP32', _mpy=10502)
```

**Comment by ricksorensen** [comment]
CPython 3.8.10 give same result returning keys as strings with `json.load()`.  `json.dump("dic",f)` writes file `dic` as;
```
$ cat dic
{"0": [0, "0"], "1": [1, "1"]}
```

See:
[https://stackoverflow.com/questions/1450957/pythons-json-module-converts-int-dictionary-keys-to-strings](https://stackoverflow.com/questions/1450957/pythons-json-module-converts-int-dictionary-keys-to-strings)

**Reply by kjm1102** [reply]
That stackoverflow was an interesting read. Coding languages remind me of junk yard dogs. No matter which one you're using they are always ready to bite you on the arse.

**Comment by peterhinch** [comment]
JSON is one of a number of serialisation solutions available in MicroPython. Each has its own benefits and drawbacks. See [this doc](https://github.com/peterhinch/micropython-samples/blob/master/SERIALISATION.md) for a review.
